train.py

```diff
@@ -401,7 +401,7 @@ def main(parsed_arg_groups: tuple[TrainingArgs, MiscArgs]):
         max_steps=args.training_goal,
         val_check_interval=args.val_frequency,
         check_val_every_n_epoch=None,  # validation based on steps instead of epochs
-        devices=args.devices,
+        devices=num_devices,    # determine the number of devices that are actually there (use one CPU if no GPUs are available)
```

Comment:
This breaks specifying `--devices=0,4,5` explicitly. But let's just make a note of it and I will fix it when I refactor the entire device handling in #32